azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -292,11 +292,31 @@ def _update(self, force_update):
         #
         if self._extensions_goal_state.source == GoalStateSource.FastTrack and self._goal_state_properties & GoalStateProperties.Certificates:
             self._check_certificates()
+            # Re-download certificates if any have been removed from disk since last download
+            if self._certs_uri is not None:
+                certificates = self.certs.summary
+                certs_missing_from_disk = False
+
+                for c in certificates:
+                    cert_path = os.path.join(conf.get_lib_dir(), c['thumbprint'] + '.crt')
+                    if not os.path.isfile(cert_path):
+                        certs_missing_from_disk = True
+                        message = "Certificate required by goal state is not on disk: (0)".format(cert_path)
+                        self.logger.info(message)
+                        add_event(op=WALAEventOperation.GoalState, message=message)
+                if certs_missing_from_disk:
+                    # Try to re-download certs. Sometimes download may fail if certs_uri is outdated/contains wrong
+                    # container id (for example, when the VM is moved to a new container after resuming from
+                    # hibernation). If download fails we should report and continue with goal state processing, as some
+                    # extensions in the goal state may succeed.
+                    try:
+                        self._download_certificates(self._certs_uri)
+                    except Exception as e:
+                        message = "Failed to download certificates: {0}".format(ustr(e))
```

Comment:
since this is a warning, let's add more context, similar to the comment above: "unable to download certificates, will continue, some extensions may fail, error: blah"

Reply:
Updated the message with more context